lib/index.js

```diff
@@ -95,10 +95,36 @@ gitUrlParse.stringify = function (obj, type) {
             return `git+ssh://git@${obj.resource}/${obj.full_name}.git`;
         case "http":
         case "https":
-            return `${type}://${obj.resource}/${obj.full_name}`;
+            let token = '';
+            if (obj.token) {
+              token = buildToken(obj);
+            }
+            return `${type}://${token}${obj.resource}/${obj.full_name}`;
         default:
             return obj.href;
     }
 };
 
+/**
```

Comment:
Use `/*!` instead of `/**`. It's a little trick to ignore the JSDoc when building the docs (we don't want to show this in the README file, since it's a private function).